dev-dotnet/dotnet-sdk/dotnet-sdk-5.0.203.ebuild

```diff
@@ -0,0 +1,224 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit check-reqs toolchain-funcs flag-o-matic multiprocessing
+
+DESCRIPTION=".NET SDK (includes runtime + aspnet)"
+HOMEPAGE="https://dotnet.microsoft.com/"
+
+BOOT_PV="5.0.202"
+BOOT_ARTIFACTS_PV="0.1.0-${BOOT_PV}-1088763-20210414.1"  # See eng/Versions.props
+BOOT_SRC_URI="https://dotnetcli.azureedge.net/dotnet/Sdk/${BOOT_PV}/dotnet-sdk-${BOOT_PV}-linux-x64.tar.gz
+	https://dotnetcli.azureedge.net/source-built-artifacts/assets/Private.SourceBuilt.Artifacts.${BOOT_ARTIFACTS_PV}.tar.gz
+"
+
+SHA512=d2a6162c62d475ea3aab0055179a405b8b99fb698c30738db4d93c6f3372da438b2c8c7ad204b56e01f1ff53a6fd1c12e118dbc7a6a6d637b4bd9392794fa529
+MY_P="dotnet-v${PV}-SDK"
+SRC_URI="https://src.fedoraproject.org/lookaside/pkgs/dotnet5.0/${MY_P}.tar.gz/sha512/${SHA512}/${MY_P}.tar.gz
+	https://src.fedoraproject.org/rpms/dotnet5.0/raw/7f6b8ec3f9527a6fbc89018ce31cdcfecc474edd/f/sdk-telemetry-optout.patch -> dotnet-sdk-telemetry-optout.patch
+	!system-bootstrap? ( ${BOOT_SRC_URI} )
+"
+S="${WORKDIR}/${MY_P}"
+
+# Using fedora source archive, since Microsoft doesn't provide one.
+# If you wish to build one yourself, see https://github.com/dotnet/source-build/
+# Do note that it isn't currently possible to generate a source archive without
+#  the proprietary "Darc CLI" tool, and downloading a bunch of nuget packages.
+
+# Fedora package script:
+# https://src.fedoraproject.org/rpms/dotnet5.0/blob/rawhide/f/build-dotnet-tarball
+# Or just run ./build-source-tarball.sh from the root of the source-build repo.
+# (the --base-tarball option might be worth using now it's added)
+
+# https://github.com/dotnet/runtime/blob/release/5.0/src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json#L27
+LICENSE="MIT Apache-2.0 BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+system-libunwind system-bootstrap"
+RDEPEND="
+	app-crypt/mit-krb5
+	dev-libs/icu
+	dev-util/lttng-ust
+	system-libunwind? ( sys-libs/libunwind )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-util/cmake
+	dev-vcs/git
+	dev-dotnet/dotnet-reference-packages
+	system-bootstrap? ( >=dev-dotnet/dotnet-sdk-${BOOT_PV} )
+"
+
+CHECKREQS_DISK_BUILD="20G"
+CHECKREQS_DISK_USR="1200M"
+
+# Use ninja unless emake is explicitly requested
+# Adapted from cmake.eclass
+case ${CMAKE_MAKEFILE_GENERATOR:-ninja} in
+	emake)
+		BDEPEND+=" sys-devel/make"
+		CMAKE_GENERATOR_FLAG=""
+		;;
+	ninja|*)
+		BDEPEND+=" dev-util/ninja"
+		CMAKE_GENERATOR_FLAG="-ninja"
+		;;
+esac
+
+# If the package can't be built using gcc currently, set this to "true"
+FORCE_CLANG="${FORCE_CLANG:-true}"
+
+if ${FORCE_CLANG}; then
+	BDEPEND+=" sys-devel/clang"
+fi
```

Comment:
I don't understand the need for this - per your comments on the PR, it doesn't work with gcc. So why not just do away with `FORCE_CLANG` variable altogether?

Reply:
I want to make it simple to disable the clang enforcement. The presence of a lot of `-gcc` switches in upstream's build scripts indicates that they at least *intend* it to work on GCC as well, so this situation might change in the future, be it through downstream or upstream patches and maintainance.

Reply:
I suppose if the intent in upstream is to work with GCC it's cool. Personally, I prefer to add these types of complexities/indirections when they're actually needed, rather than planning for some eventual future need.

However if it is easier to do this now rather then later, then it definitely makes sense.